client/src/main/java/io/pravega/client/stream/impl/EventStreamWriterImpl.java

```diff
@@ -79,9 +80,11 @@
     private final SegmentSelector selector;
     private final Consumer<Segment> segmentSealedCallBack;
     private final ExecutorService retransmitPool;
+    private final TxnKeepAlive txnKeepAlive;
     
     EventStreamWriterImpl(Stream stream, Controller controller, SegmentOutputStreamFactory outputStreamFactory,
-            Serializer<Type> serializer, EventWriterConfig config, ExecutorService retransmitPool) {
+                          Serializer<Type> serializer, EventWriterConfig config, ExecutorService retransmitPool,
+                          ConnectionFactory connectionFactory) {
```

Comment:
This feels a bit hack-ish as we really only need a single thread to be doing the pinging periodically. Is it worth not making this change and having a single thread (or a singleton thread pool) in `TxnKeepAlive` that is created and closed there?

Reply:
ok, updated the code.